### ISSUE TITLE
Strange behavior of results delivered from earthworm server

### ISSUE THREAD

**Issue by megies** [obspy/obspy#1109]
It seems that sometimes results of earthworm servers are formatted differently, such that the "result flag" we are checking is on a different position, resulting in strange error messages etc.

This looks like it is probably an issue in the server (esp. since the strange duplication of "`rwserv`" at the start), so we should contact somebody from @usgs who can comment on this. @seismopaul, can you comment on this maybe? Anybody else who might know more about the earthworm server inner workings?

(Problem is the same on Python2/3)

This is where we check the result of the request:
https://github.com/obspy/obspy/blob/releases/obspy/earthworm/waveserver.py#L253

But sometimes the result has another item at the front so that the result flag is not looked up correctly (print statement not in original source code):

``` python
from obspy import UTCDateTime
from obspy.earthworm import Client
client = Client("pubavo1.wr.usgs.gov", 16022, timeout=30.0)
t = UTCDateTime() - 1000
print(client.getWaveform('AV', 'ACH', '--', 'EH*', t, t+10))
print(client.getWaveform('AV', 'ACH', '*', 'EH*', t, t+10))
```

```
rwserv 74 ACH EHZ AV -- F i2 1.4393765432900329E9 1.439376554290032E9 2904
['rwserv', '74', 'ACH', 'EHZ', 'AV', '--', 'F', 'i2', '1.4393765432900329E9', '1.439376554290032E9', '2904']
rwserv 76 ACH EHN AV -- F i2 1.4393765432900329E9 1.439376554290032E9 2904
['rwserv', '76', 'ACH', 'EHN', 'AV', '--', 'F', 'i2', '1.4393765432900329E9', '1.439376554290032E9', '2904']
rwserv 239 ACH EHE AV -- F i2 1.4393765432900329E9 1.439376554290032E9 2904
['rwserv', '239', 'ACH', 'EHE', 'AV', '--', 'F', 'i2', '1.4393765432900329E9', '1.439376554290032E9', '2904']
3 Trace(s) in Stream:
AV.ACH..EHZ | 2015-08-12T10:49:03.320033Z - 2015-08-12T10:49:13.320033Z | 100.0 Hz, 1001 samples
AV.ACH..EHN | 2015-08-12T10:49:03.320033Z - 2015-08-12T10:49:13.320033Z | 100.0 Hz, 1001 samples
AV.ACH..EHE | 2015-08-12T10:49:03.320033Z - 2015-08-12T10:49:13.320033Z | 100.0 Hz, 1001 samples
rwserv rwserv 0 ACH EHZ AV * FN
['rwserv', 'rwserv', '0', 'ACH', 'EHZ', 'AV', '*', 'FN']
Traceback (most recent call last):
  File "/tmp/ewtest.py", line 6, in <module>
    print(client.getWaveform('AV', 'ACH', '*', 'EH*', t, t+10))
  File "/home/megies/git/obspy/obspy/earthworm/client.py", line 105, in getWaveform
    cleanup=cleanup)
  File "/home/megies/git/obspy/obspy/earthworm/client.py", line 112, in getWaveform
    timeout=self.timeout)
  File "/home/megies/git/obspy/obspy/earthworm/waveserver.py", line 258, in readWaveServerV
    print(msg % (flag, RETURNFLAG_KEY[flag]))
KeyError: '*'
```

print statements included into source to show the issue:

``` python
  247     sock = sendSockReq(server, port, reqstr.encode('ascii', 'strict'),             
  248                        timeout=timeout)                                            
  249     r = getSockCharLine(sock, timeout=timeout)                                     
  250     if not r:                                                                      
  251         return []                                                                  
  252     tokens = str(r.decode()).split()                                               
  253     flag = tokens[6]                                                               
+ 254     print(r.decode())
+ 255     print(tokens)
  256     if flag != 'F':                                                                
  257         msg = 'readWaveServerV returned flag %s - %s'                              
  258         print(msg % (flag, RETURNFLAG_KEY[flag]))                                  
  259         return []                                                                  
```

**Comment by megies** [comment]
Here's the documentation of the expected result (at the end, "GETSCNLRAW"):
http://folkworm.ceri.memphis.edu/ew-doc/PROGRAMMER/wsv_protocol.html

`'rwserv'` is actually what we provide (hardcoded) as "request id" and it should not get duplicated in the response, so this is definitely a server side issue.

**Comment by tparker-usgs** [comment]
This behavior was caused by a bug in the Winston wave server, a popular server which implements the Earthworm WSV protocol. pubavo1.wr.usgs.gov used in the example here is a Winston wave server. 

That bug has been fixed in Winston version 1.6.2. It'll take time for servers to be upgraded., the specific server mentioned here is scheduled to be upgraded to the new version next week. 

I'm closing this ticket as the problem has been corrected upstream and no changes are needed in ObsPy.

**Comment by megies** [comment]
Thanks for the feedback @tparker-usgs! :+1:
